Sources/Valet/Internal/Keychain.swift

```diff
@@ -337,64 +276,63 @@ internal final class Keychain {
             }
             
             guard !key.isEmpty else {
-                return .keyInQueryResultInvalid
+                throw MigrationError.keyInQueryResultInvalid
             }
             
             guard !keysToMigrate.contains(key) else {
-                return .duplicateKeyInQueryResult
+                throw MigrationError.duplicateKeyInQueryResult
             }
             
             guard let data = keychainEntry[kSecValueData as String] as? Data, !data.isEmpty else {
-                return .dataInQueryResultInvalid
+                throw MigrationError.dataInQueryResultInvalid
             }
-            
-            guard case let .error(status) = Keychain.containsObject(forKey: key, options: destinationAttributes), status == errSecItemNotFound else {
-                return .keyInQueryResultAlreadyExistsInValet
+
+            if Keychain.containsObject(forKey: key, options: destinationAttributes) == errSecItemNotFound {
+                keysToMigrate.insert(key)
+            } else {
+                throw MigrationError.keyInQueryResultAlreadyExistsInValet
```

Comment:
Should this throw a different error depending on what `containsObject(...)` returns?

Reply:
I don't think so? Note that we're currently preserving existing behavior here (we previously returned a `keyInQueryResultAlreadyExistsInValet`, but now we're throwing). If we want to update this logic, let's do that in a separate PR 🙂